src/Umbraco.Cms.Api.Delivery/Services/ApiMediaQueryService.cs

```diff
@@ -66,7 +66,8 @@ private IPublishedMediaCache GetRequiredPublishedMediaCache()
     private IPublishedContent? TryGetByPath(string path, IPublishedMediaCache mediaCache)
     {
         var segments = path.Split(Constants.CharArrays.ForwardSlash, StringSplitOptions.RemoveEmptyEntries);
-        IEnumerable<IPublishedContent> currentChildren = mediaCache.GetAtRoot();
+        _mediaNavigationQueryService.TryGetRootKeys(out IEnumerable<Guid> rootKeys);
+        IEnumerable<IPublishedContent> currentChildren = rootKeys.Select(x => mediaCache.GetById(false, x)).WhereNotNull();
```

Comment:
Seems like extracting `rootKeys.Select(x => mediaCache.GetById(false, x)).WhereNotNull()` into a prvate helper method might be useful, since we do similar on line 107.